src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -304,6 +304,9 @@ public async Task<SyncTriggersPayload> GetSyncTriggersPayload()
             var functionDetails = await WebFunctionsManager.GetFunctionMetadataResponse(listableFunctions, hostOptions, _hostNameProvider);
             result.Add("functions", new JArray(functionDetails.Select(p => JObject.FromObject(p))));
 
+            // Add host.json to the pyaload
+            result.Add("host.json", GetHostJson());
```

Comment:
Note that the extended format { triggers: [], functions: [], secrets: {} } is best effort, and if the payload is oversize based on what FE can handle, we revert to the old triggers array format. This is fine because existing usages of the additional info are able to handle this - the extra data just enables optimizations.

However, in your case, it seems the host config level trigger info you want to add isn't optional? I.e. what what happen in your scenarios if due to size the payload reverted to minimal format?

Reply:
Also I worry about dumping the entire host.json file contents. Aren't we really only interested in host level extensions config?

Reply:
Thank you @mathewc for the review. I didn't know the restriction. 
I updated the PR. We deicide to narrow it down the scope. Our usage is, under the Kubernetes Environment, The Sync Trigger will post request to the Kudu Lite API. (As far as I know, It doesn't limit the payload size).  In this case, looks no problem? I'm not sure which part is required for Durable/Logic Apps usage. @cgillum  Do you know which part is required?

Reply:
@mathewc It sounds like you're saying we can't rely on this logic to actually work? What controls whether we use the extended format or the "old triggers array format"? Also, what is the maximum size allowed by the FE? Indeed, the host.json metadata is required for our scenarios. We could probably trim it down to just the "extensions" section, though, if you're worried about too much host.json bloat.

Reply:
The code [here](https://github.com/Azure/azure-functions-host/blob/d831b969dcd3c8ed6a6949b82329b0db7ababf12/src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs#L349) enforces the FE size limit. What we'd have to do is redefine what the "minimal format" is, and rather than just returning the triggers [] in such cases, we'd return `{ triggers: [], extensions: [] }`, scoping it so that we're not dumping the entire host.json in there, but only the extensions config section.

Reply:
Also note that there are still some legacy codepaths in Kudu (e.g. [here](https://github.com/projectkudu/kudu/blob/0fbbb59fd1151f61daee544715e1326a8632be76/Kudu.Core/Helpers/PostDeploymentHelper.cs#L266) that will end up doing an "old format" sync). @ankitkumarr did the work in Kudu to forward old Kudu synctriggers calls to the Functions runtime to get the new format, but if that fails for any reason, Kudu falls back to it's previous logic where the sync formatting was done by Kudu. So any consumers of this synctriggers payload (ScaleController, Geo ARM APIs, etc.) need to handle both formats, and they do currently.

Reply:
Thank you  @mathewc , @cgillum @ankitkumarr 
Since I create this PR for unblock the Durable usage,  Is this looks good? or any idea/discussion?

1. limit the payload adding `extensions` part only for kubernetes environmnet.
2. Only the kubernetes environmnet, disable the MaxTriggersStringLength (it is 204800)
3. Create an issue for make SyincTrigger with host.json payload in global for keep on discussing it. 

Only thing, what I don't know is the [code](https://github.com/projectkudu/kudu/blob/0fbbb59fd1151f61daee544715e1326a8632be76/Kudu.Core/Helpers/PostDeploymentHelper.cs#L266) that is done by @ankitkumarr  is releated kubernets environmnet. We are working on this [branch](https://github.com/Azure-App-Service/KuduLite/tree/feature/k8se)